### DISCUSSION TITLE
Suggestions for improvement

### DISCUSSION THREAD

**Discussion by FilipDem** [jvanderzande/GarbageCalendar#26] [Ideas]
Hi,
Thanks for the good script.
Seen the text mentioned in the text device, the size of the device goes beyond the standard size. This gives a small visual negative effect:

This could be improved by the possibility to change the font size (I hardcoded it, but could be included in the configuration file).
      commandArray['UpdateDevice'] = otherdevices_idx[myGarbageDevice] .. '|0|' .. '<font size="0">' .. devtxt .. '</font>'

Another improvement could be the following. Often different garbage collections are done at the same day (eg glass and carton, or GFT and fraction)... So why using different lines for this? Wouldn't it be rather interesting to inverse the logic: having first the date and give a list of types of garbage collections.
So for instance:
maandag x september: Glas, Papier
woensdag y september: Restafval

Thanks,

Filip

**Comment by jvanderzande** [comment]
> Seen the text mentioned in the text device, the size of the device goes beyond the standard size. This gives a small visual negative effect:
> This could be improved by the possibility to change the font size (I hardcoded it, but could be included in the configuration file).
> commandArray['UpdateDevice'] = otherdevices_idx[myGarbageDevice] .. '|0|' .. '' .. devtxt .. ''
> 

Have you tried to add any formatting you like to the textformat value in GarbageConfig.lua? 
For example:
`
textformat = '<font color="red" size=1>tdesc: wd dd mmm</font>'
`

> Another improvement could be the following. Often different garbage collections are done at the same day (eg glass and carton, or GFT and fraction)... So why using different lines for this? Wouldn't it be rather interesting to inverse the logic: having first the date and give a list of types of garbage collections.
> So for instance:
> maandag x september: Glas, Papier
> woensdag y september: Restafval

I am trying to understand the benefit of this or what actually would change as far as notifications etc?
To me it would take away all flexibility of changes made to the schedule by the garbage collecting company when you assume it is always on a single day, but maybe I am not totally understanding your thought on how this would work.
Also defining the alternative **Text** for each Garbage type would be a challenge with this change...right?

Jos

**Comment by FilipDem** [comment]
OK, thanks... The first part works also... Thanks for the feedback.
For the second topic... Notifications etc should not change... It is only cosmetic... Just to get less lines... When several 'collections' are on the same day, why not having it on one line.
Could also be in your order (first the type and then the date could stay).
Glas, Papier: maandag x september
Restafval: woensdag y september
Was just a suggestion...

**Reply by jvanderzande** [reply]
No worries, I appreciate the input/questions, just like to understand what you are proposing first.
I guess I misunderstood your first post and assume now that you are talking about the output shown in the Domoticz Device and not about the Config for each Garbage type...right?  So you like the **_tdesc_** to contain all garbagetypes for that day in stead of one for each garagetype when occurring on the same day.  This would require even more space in the Domoticz device but assume that could be doable with a config parameter.

**Comment by FilipDem** [comment]
Yes indeed I was talking about the Domoticz Device... Because this device is the only that is "bigger" (in size) than the other ones... Because of the lines of text...
You should not do it only for me... If you think it is interesting, you can do it...

**Reply by jvanderzande** [reply]
I will have a look and make it optional to combine GarbageTypes for the same day on one line...  shouldn't be too hard and I can see the use of this allowing the next X days of garbage collection to be shown in instead of the next X GarbageTypes. 
Will commit something in the development branch soon so you can test it.

**Comment by FilipDem** [comment]
Top!

Op zo 29 aug. 2021 14:33 schreef Jos ***@***.***>:

**Reply by jvanderzande** [reply]
Just committed an update to development for the main script. Add the last line (Combine_Garbage_perDay )  to you garbagecalendarconfig.lua:
```
ShowNextEvents = 3               -- indicate the next x eventlines to show in the TEXT Sensor in Domoticz
Combine_Garbage_perDay = true   -- (default) false will show each garbagetype on its own line
                                 -- true will show multiple garbagetype on a single line when collected the same day
```

Let me know if this is what you want. 
PS: The format you shown in your initial post can be set by updating the **_textformat_** option. :)

**Reply by jvanderzande** [reply]
Did some more testing today and added some fixes in the last commit for the ShowSinglePerType option.

**Comment by FilipDem** [comment]
Hi Jos,
Thanks for the information and for your work.
I just want to let you know that I didn't tried yet. I will probably do it the weekend 11/09 and will give you feedback (I am not at home the next 2 weeks).

About the format I already applied (and gave feedback in the discussion earlier) and this works indeed.

Keep you posted!

**Reply by jvanderzande** [reply]
Hope the vacation was good :) ....  Did you have a chance to try?

**Comment by FilipDem** [comment]
Hi Jos,
Yes... Initially I gave feedback, but afterwards I saw it was in the wrong
project :-)...
All is running well, unfortunately there is now the case that for the next
garbage collections there are all on different dates. So I haven't seen yet
working that they are grouped.
If there would be sth wrong, I will let you know.
Filip

Op ma 20 sep. 2021 om 14:45 schreef Jos ***@***.***>:
